test/sql/window/test_mode_window.test

```diff
@@ -0,0 +1,63 @@
+# name: test/sql/window/test_mode_window.test
+# description: Test MODE aggregate as a window function
+# group: [window]
+
+require vector_size 512
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+create table modes as select range r from range(10) union all values (NULL), (NULL), (NULL);
+
+query IIII
+SELECT r % 2, r, r/3, mode(r/3) over (partition by r % 2 order by r) FROM modes ORDER BY 1, 2
+----
+NULL	NULL	NULL	NULL
+NULL	NULL	NULL	NULL
+NULL	NULL	NULL	NULL
+0	0	0	0
+0	2	0	0
+0	4	1	0
+0	6	2	0
+0	8	2	0
+1	1	0	0
+1	3	1	0
+1	5	1	1
+1	7	2	1
+1	9	3	1
+
+query III
+SELECT r, r/3, mode(r/3) over (order by r rows between 1 preceding and 1 following) FROM modes ORDER BY 1, 2
+----
+NULL	NULL	NULL
+NULL	NULL	NULL
+NULL	NULL	0
+0	0	0
+1	0	0
+2	0	0
+3	1	1
+4	1	1
+5	1	1
+6	2	2
+7	2	2
+8	2	2
+9	3	2
+
+query III
+SELECT r, r/3, mode(r/3) over (order by r rows between 1 preceding and 3 following) FROM modes ORDER BY 1, 2
```

Comment:
* Could we add a test with a bunch of scattered null values? e.g. you would order by column a, perform the quantile on column b, and column b is basically 50/50 null/not null.
* Perhaps a test with UNBOUNDED PRECEDING and a test with UNBOUNDED FOLLOWING? Perhaps also a test with both?

Reply:
This is `MODE` but yes we should hit them both.

Reply:
These tests also drove a small bug in `Finalize` for `MODE` when the hash table is empty.